Sources/Core/Sources/Render/ScreenRenderer.swift

```diff
@@ -0,0 +1,131 @@
+import Foundation
+import MetalKit
+
+/// The renderer managing offscreen rendering and displaying final result in view.
+public final class ScreenRenderer: Renderer {
+  // MARK: - Private properties
+  /// The device used to render.
+  private var device: MTLDevice
+  
+  /// Renderer's own pipeline state (for drawing on-screen)
+  private var pipelineState: MTLRenderPipelineState
+  
+  /// Offscreen render pass used to command rendering into renderer's internal textures
+  private var offscreenRenderPass: MTLRenderPassDescriptor!
```

Comment:
Rename to `offScreenRenderDescriptor` or `offScreenRenderPassDescriptor` or something like that to make it clear that it's a descriptor. And I think `offScreen` is the correct capitalisation, but if it's `offscreen` use that instead